src/wristpy/core/orchestrator.py

```
@@ -0,0 +1,259 @@
+            the timestamps in the reference_measure.
+
+    """
+    nonwear_df = pl.DataFrame(
+        {
+            "nonwear": nonwear_data.measurements.astype(np.int64),
+            "time": nonwear_data.time,
+        }
+    ).set_sorted("time")
+
+    nonwear_upsample = nonwear_df.upsample(time_column="time", every="5s").fill_null(
+        strategy="forward"
+    )
+
+    end_sequence = pl.repeat(
+        nonwear_upsample["nonwear"][-1],
+        n=(len(reference_measure.time) - len(nonwear_upsample["time"])),
+        dtype=pl.Int64,
+        eager=True,
+    )
+    full_nonwear_array = pl.concat([nonwear_upsample["nonwear"], end_sequence])
+
+    return full_nonwear_array.to_numpy()
+
+
+def run(
+    input: pathlib.Path,
+    output: Optional[pathlib.Path] = None,
+    settings: config.Settings = config.Settings(),
+    calibrator: calibration.Calibration = calibration.Calibration(),
+    detect_nonwear_kwargs: Optional[Dict[str, Any]] = None,
+) -> Results:
+    """Runs wristpy.
+
+    Args:
+        input: Path to the input file to be read. Currently supports .bin and .gt3x
+        output: Path to save data to. The path should end in the file name to be given
+            to the save data. Two files will be saved, each with the given file name and
+            the _raw_time or _epoch1 label after. Currently supports saving in .csv and
+            .parquet
+        settings: The settings object from which physical activity levels are taken.
+        calibrator: The calibrator to be used on the input data.
+        detect_nonwear_kwargs: The arguments to the nonwear function delivered as a
+            dictionary. Arguements are short_epoch_length, n_short_epoch_in_long_epoch,
+            std_criteria, range_criteria.
+
+    Returns:
+        All calculated data in a save ready format as a Results object.
+
+    """
+    watch_data = readers.read_watch_data(input)
```

When dealing with user input, fail early. We can already check at the start whether the output extension is in a list of accepted extensions.